extensions/amp-script/amp-script.md

```diff
@@ -46,61 +44,101 @@ Allows rendering of custom UI components running on third-party JavaScript.
-  button.addEventListener('click', () => {
-    const h1 = document.createElement('h1');
-    h1.textContent = 'Hello World!';
-    // `document.body` is effectively the <amp-script> element.
-    document.body.appendChild(h1);
+<script id="hello-world" type="text/plain" target="amp-script">
+  const btn = document.querySelector('button');
+  btn.addEventListener('click', () => {
+    document.body.textContent += 'Hello World!';
   });
 </script>
 ```
 
+{% call callout('Tip', type='success') %}
+Enable the experiment via `AMP.toggleExperiment('amp-script')` in dev console.
+{% endcall %}
+
+### How does it work?
+
+`amp-script` runs your custom JavaScript in a [Web Worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers) that contains a virtual DOM. When your JavaScript code modifies this virtual DOM, `amp-script` forwards these changes to the main thread and applies them to the `amp-script` element subtree.
+
+For example, adding an element to `document.body`:
+
+```js
+// my-script.js
+const p = document.createElement('p');
+p.textContent = 'I am added to the body!';
+document.body.appendChild(p);
+```
+
+Will be reflected on the page as a new child of the `amp-script` element:
+
 ```html
-<!-- Using an remote script ("src" attribute). -->
-<amp-script layout="container" src="https://example.com/hello-world.js">
-  <button id="hello">Insert Hello World!</button>
+<amp-script src="http://example.com/my-script.js" width=300 height=100>
+  <p>I am added to the body!</p>
 </amp-script>
 ```
 
-{% call callout('Tip', type='success') %}
-Enable the experiment via `AMP.toggleExperiment('amp-script')` in dev console.
-{% endcall %}
+Under the hood, `amp-script` uses [@ampproject/worker-dom](https://github.com/ampproject/worker-dom/). For design details, see the ["Intent to Implement" issue](https://github.com/ampproject/amphtml/issues/13471).
 
-For additional code samples, see [`examples/amp-script/`](https://github.com/ampproject/amphtml/tree/master/examples/amp-script).
+### Restrictions
 
-### How does it work?
+#### Size of JavaScript code
 
-`amp-script` is an AMP component wrapper for [`worker-dom`](https://github.com/ampproject/worker-dom/).
+`amp-script` has the following restrictions on JavaScript file size:
 
-`worker-dom` runs third-party JavaScript in a web worker containing a virtual DOM. The virtual DOM listens for mutations and forwards them to the main page which reflects the changes on the real DOM.
+- Maximum of 10,000 bytes per `amp-script` element that uses a local script via `script[type=text/plain][target=amp-script]`.
+- Maximum total of 150,000 bytes for all `amp-script` elements on the page.
 
-For design details, see the ["Intent to Implement" issue](https://github.com/ampproject/amphtml/issues/13471).
-For more information on `worker-dom`, see the [@ampproject/worker-dom](https://github.com/ampproject/worker-dom/) repository.
+#### Required user gestures
```

Comment:
Are user gestures required? Have we removed the [300px hight exception?](https://amp.dev/documentation/guides-and-tutorials/develop/custom-javascript?format=websites#unprompted-changes)

Reply:
It's (3) below: https://github.com/ampproject/amphtml/pull/23221/files#diff-1e9abb643c017d697cacb9e505f22839R145

Changed this to just "User gestures" since it's under "Restrictions" anyways.

Reply:
Haha, my apologies - I totally just missed this